Production ready, canonical code (#1)
singpass/myinfo-connector-dotnet: README.md

```

### 1.3 Configuration file

Config can be supplied in a number of ways, app.config, NameValueCollection or by passing a `Func<string,string>`.

You are required to create a config file with the following key values for this library. Sample config files can be found in this repository under the Sample Configuration folder.

| Required config values | Description |
| -------- | ----------- |

> | PrivateCertificateFilename | Path of the .p12 file that holds the private key. Absolute or relative path to assembly |

Reviewer: If we are going to rename this then `PrivateKeyFilename` may be more appropriate as private certificate can also mean SSL certificate issued by private CA.
```suggestion
| PrivateKeyFilename | Path of the .p12 file that holds the private key. Absolute or relative path to assembly |
```

> | PrivateCertificatePassword | Password of your private key p12 file. |

Reviewer: Related to above.
```suggestion
| PrivateKeyPassword | Password of your private key p12 file. |
```

| PublicCertificateFilename | Path to the MyInfo public certificate. Absolute or relative path to assembly  |

> | ClientAppId | Unique ID provided upon approval of your application to use MyInfo. For our sample application, it is **STG2-MYINFO-SELF-TEST** |

Reviewer: Referencing OAuth, will prefer to stick to `ClientId`.
```suggestion
| ClientId | Unique ID provided upon approval of your application to use MyInfo. For our sample application, it is **STG2-MYINFO-SELF-TEST** |
```

> | ClientAppPassword | Secret key provided upon approval of your application to use MyInfo. For our sample application, it is **44d953c796cccebcec9bdc826852857ab412fbe2** |

Reviewer: Referencing OAuth, will prefer to stick to `ClientSecret`.
```suggestion
| ClientSecret | Secret key provided upon approval of your application to use MyInfo. For our sample application, it is **44d953c796cccebcec9bdc826852857ab412fbe2** |
```

| AttributeCsv | Comma separated list of attributes requested. Possible attributes are listed in the Person object definition in the API specifications. |
| Environment | The environment your application is configured. This can be **SANDBOX**, **TEST** or **PROD**. |
| AuthoriseUrl | Specify the AUTHORISE API URL for MyInfo. The API is available in three environments:<br> SANDBOX: **https://sandbox.api.myinfo.gov.sg/com/v3/authorise**<br> TEST: **https://test.api.myinfo.gov.sg/com/v3/authorise**<br> PROD:  **https://api.myinfo.gov.sg/com/v3/authorise** |
| TokenUrl | Specify the TOKEN API URL for MyInfo. The API is available in three environments:<br> SANDBOX: **https://sandbox.api.myinfo.gov.sg/com/v3/token**<br> TEST: **https://test.api.myinfo.gov.sg/com/v3/token**<br> PROD:  **https://api.myinfo.gov.sg/com/v3/token** |
| PersonUrl | Specify the PERSON API URL for MyInfo. The API is available in three environments:<br> SANDBOX: **https://sandbox.api.myinfo.gov.sg/com/v3/person**<br> TEST: **https://test.api.myinfo.gov.sg/com/v3/person**<br> PROD:  **https://api.myinfo.gov.sg/com/v3/person** |

> | Purpose | The text passed to the OAuth to inform the user what 

Reviewer: ```suggestion
| Purpose | The text passed to the OAuth service to inform the user about the purpose to get his/her data. This will be shown to the user when requesting for his/her consent.
```

## How to use the connector

### 1. Create an instance of the MyInfoConnector

```
Func<string, string> getConfig = key => 
{ 
	// given a key, return the value. This pattern allows external methods of encryption/decryption of secrets
};
var connector = MyInfoConnector.Create(getConfig);
```

> ### 2. Construct the URL that the user should be redirected to 

Reviewer: ```suggestion
### 2. Construct the authorise URL 
The authorise URL includes the redirect URL so that the service can redirect the user to it after granting consent.
```

```
var authoriseUrl = connector.GetAuthoriseUrl();

> // Receive the callback and get the `authcode`

Reviewer: ```suggestion
var authoriseUrl = connector.GetAuthoriseUrl(redirectUrl);
// Invoke the authorise endpoint with authoriseUrl. Receive the callback and get the `authcode`
```

```

> ### 3. Retrieve the persons data

Reviewer: ```suggestion
### 3. Retrieve the person's data
```
